src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2027,12 +2027,75 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
                 Array.Clear(tmpNestedModules, 0, tmpNestedModules.Length);
             }
 
-            // Set the private data member for the module if the manifest contains
-            // this member
-            object privateData = null;
-            if (data.Contains("PrivateData"))
+            // Set the private data member for the module if the manifest contains this member
+            object privateData = data["PrivateData"];
+
+            // Validate the 'ExperimentalFeatures' member of the manifest
+            List<ExperimentalFeature> expFeatureList = null;
+            if (privateData is Hashtable hashData && hashData["PSData"] is Hashtable psData)
             {
-                privateData = data["PrivateData"];
+                if (!GetScalarFromData(psData, moduleManifestPath, "ExperimentalFeatures", manifestProcessingFlags, out Hashtable[] features))
+                {
+                    containedErrors = true;
+                    if (bailOnFirstError) return null;
+                }
+
+                if (features != null && features.Length > 0)
+                {
+                    bool nameMissingOrEmpty = false;
+                    var invalidNames = new List<string>();
+                    string moduleName = ModuleIntrinsics.GetModuleName(moduleManifestPath);
+                    expFeatureList = new List<ExperimentalFeature>(features.Length);
+
+                    foreach (var feature in features)
+                    {
+                        string featureName = feature["Name"] as string;
+                        if (String.IsNullOrEmpty(featureName))
+                        {
+                            nameMissingOrEmpty = true;
+                            continue;
+                        }
+
+                        if (ExperimentalFeature.IsModuleFeatureName(featureName, moduleName))
+                        {
+                            string featureDescription = feature["Description"] as string;
+                            expFeatureList.Add(new ExperimentalFeature(featureName, featureDescription, moduleManifestPath,
+                                                                       ExperimentalFeature.HasEnabled(featureName)));
+                        }
+                        else
```

Comment:
Given the `continue` above, maybe it's more consistent to use another one here? As in, reverse the check, so it's:
```csharp
if (!ExperimentalFeature.IsModuleFeatureName(featureName, moduleName))
{
    invalidNames.Add(featureName);
    continue;
}
```

Reply:
Will remove the `continue`, and use `if (string.IsNullOrEmpty()... elseif... else...)`